hoomd/md/compute.py

```diff
@@ -0,0 +1,344 @@
+# Copyright (c) 2009-2019 The Regents of the University of Michigan
+# This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+# Maintainer: joaander / All Developers are free to add commands for new features
+
+""" Compute system properties """
+
+from hoomd import _hoomd;
+from hoomd.md import _md
+from hoomd.operation import _Compute
+from hoomd.logging import log
+import hoomd;
+import sys;
+
+class _Thermo(_Compute):
+
+    def __init__(self, filter):
+        self._filter = filter
+
+
+class ThermodynamicQuantities(_Thermo):
+    """ Compute thermodynamic properties of a group of particles.
+
+    Args:
+        filter (``hoomd.filter``): Particle filter to compute thermodynamic properties for.
+
+    :py:class:`ThermodynamicQuantities` acts on a given group of particles and calculates thermodynamic properties of
+    those particles when requested. All specified :py:class:`ThermodynamicQuantities` objects can be added to a logger
+    for logging during a simulation, see :py:class:`hoomd.logging.Logger` for more details.
+
+    Note:
+        For pairwise terms :math:`U_{ij}` and :math:`F_{ij} \\cdot r_{ij}`, HOOMD-blue assigns half of each term to a
+        per-particle quantity. When filter selects a subset of the particles in the system, :py:class:`ThermodynamicQuantities`
+        sums these per-particle quantities over only particles selected by the filter.
+
+    Examples::
+
+        f = filter.Type('A')
+        compute.ThermodynamicQuantities(filter=f)
+    """
+
+    def __init__(self, filter):
+        super().__init__(filter)
+
+    def _attach(self):
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            thermo_cls = _md.ComputeThermo
+        else:
+            thermo_cls = _md.ComputeThermoGPU
+        group = self._simulation.state.get_group(self._filter)
+        self._cpp_obj = thermo_cls(self._simulation.state._cpp_sys_def, group, "")
+        super()._attach()
+
+    @log
+    def kinetic_temperature(self):
+        """
+        :math:`kT`, instantaneous thermal energy of the group (in energy units).
+
+        Calculated as:
+
+          .. math::
+
+            kT = 2 \\cdot \\frac{K}{N_{\\mathrm{dof}}}
+        """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.kinetic_temperature
+        else:
+            return None
+
+    @log
+    def pressure(self):
+        """
+        :math:`P`, instantaneous pressure of the group (in pressure units).
+
+        Calculated as:
+
+        .. math::
+
+            P = \\left(\\frac{2}{D} \\cdot K_{\\mathrm{trans}} + \\frac{1}{2} \\cdot W \\right) / V,
```

Comment:
```suggestion
            P = \\left(\\frac{2}{D} \\cdot K_{\\mathrm{trans}} + \\frac{1}{D} \\cdot W \\right) / V,
```
Or factor out the 1/D.